type-definitions/ts-tests/map.ts

```diff
@@ -1,631 +1,673 @@
+import { expectType, expectAssignable, expectError } from 'tsd';
 import { Map, List, MapOf } from 'immutable';
 
 {
   // #constructor
 
-  // $ExpectType Map<unknown, unknown>
-  Map();
+  expectType<Map<unknown, unknown>>(Map());
 
-  // $ExpectType Map<number, number>
-  Map<number, number>();
+  expectType<Map<number, number>>(Map<number, number>());
 
-  // $ExpectType Map<number, string>
-  Map([[1, 'a']]);
+  expectType<Map<number, string>>(Map([[1, 'a']]));
 
-  // $ExpectType Map<string, string>
-  Map([['a', 'a']]);
+  expectType<Map<string, string>>(Map([['a', 'a']]));
 
-  // $ExpectType Map<number, string>
-  Map(List<[number, string]>([[1, 'a']]));
+  expectType<Map<number, string>>(Map(List<[number, string]>([[1, 'a']])));
 
-  // $ExpectType MapOf<{ a: number; }>
-  Map({ a: 1 });
+  expectType<MapOf<{ a: number }>>(Map({ a: 1 }));
 
-  // $ExpectType MapOf<{ a: number; b: string; }>
-  Map({ a: 1, b: 'b' });
+  expectType<MapOf<{ a: number; b: string }>>(Map({ a: 1, b: 'b' }));
 
-  // $ExpectType MapOf<{ a: MapOf<{ b: MapOf<{ c: number; }>; }>; }>
-  Map({ a: Map({ b: Map({ c: 3 }) }) });
+  expectType<MapOf<{ a: MapOf<{ b: MapOf<{ c: number }> }> }>>(
+    Map({ a: Map({ b: Map({ c: 3 }) }) })
+  );
 
-  // $ExpectError
-  Map<{ a: string }>({ a: 1 });
+  expectError(Map<{ a: string }>({ a: 1 }));
 
-  // $ExpectError
-  Map<{ a: string }>({ a: 'a', b: 'b' });
+  expectError(Map<{ a: string }>({ a: 'a', b: 'b' }));
 
   // No longer works in typescript@>=3.9
-  // // $ExpectError - TypeScript does not support Lists as tuples
-  // Map(List([List(['a', 'b'])]));
+  // TypeScript does not support Lists as tuples
+  // expectError(Map(List([List(['a', 'b'])])));
 
-  // $ExpectType Map<number, number>
-  const numberMap: Map<number, number> = Map();
+  expectAssignable<Map<number, number>>(Map());
```

Comment:
Just to draw attention, here you are hitting one of `tsd`s oddities. Both of the following tests are passing:

```ts
expectAssignable<Map<number, number>>(Map());
expectNotAssignable<Map<number, number>>(Map());
```

Reply:
Just above there is this test:

```ts
expectType<Map<unknown, unknown>>(Map())
```

If the inferred type of `Map()` is `Map<unknown, unknown>` it cannot be assigned to `Map<number, number>`.

To make it assignable, the type has to be cast to `Map<number, number>` first. This is what `expectAssignable()` does, but `expectNotAssignable()` does not. In my opinion `expectNotAssignable()` behaves correctly.